pkg/allocateip/allocateip_test.go

```diff
@@ -435,6 +435,36 @@ var _ = Describe("determineEnabledPoolCIDRs", func() {
 			Expect(cidrs).To(ContainElement(*cidr1))
 			Expect(cidrs).ToNot(ContainElement(*cidr2))
 		})
+		It("should match ip-pool-1 but not ip-pool-2 for VXLANMode CrossSubnet", func() {
+			// Mock out the node and ip pools
+			n := api.Node{ObjectMeta: metav1.ObjectMeta{Name: "bee-node", Labels: map[string]string{"foo": "bar"}}}
+			pl := api.IPPoolList{
+				Items: []api.IPPool{
+					api.IPPool{
+						ObjectMeta: metav1.ObjectMeta{Name: "ip-pool-1"},
+						Spec: api.IPPoolSpec{
+							Disabled:     false,
+							CIDR:         "172.0.0.0/9",
+							NodeSelector: `foo == "bar"`,
+							VXLANMode:    api.VXLANModeCrossSubnet,
+						},
+					}, api.IPPool{
+						ObjectMeta: metav1.ObjectMeta{Name: "ip-pool-2"},
+						Spec: api.IPPoolSpec{
+							Disabled:     false,
+							CIDR:         "172.128.0.0/9",
+							NodeSelector: `foo != "bar"`,
+							VXLANMode:    api.VXLANModeCrossSubnet,
+						},
+					}}}
+
+			// Execute and test assertions.
+			cidrs := determineEnabledPoolCIDRs(n, pl, true)
```

Comment:
I don't understand this test fully. I think it's testing the `NodeSelector` field rather than the `VXLANModeCrossSubnet` field, IIUC.

I think we want two pools which select this node - one with no VXLAN and one with CrossSubnet, and then assert that the CrossSubnet one is selected?

Reply:
The test is to make sure that a VXLAN pool with VXLANMode CrossSubnet is selected in the same way the Always is. So this tests that a matching CrossSubnet pool is selected, and that a none matching CrossSubnet pool is not (just to make sure the does not match logic still works for CrossSubnet)